makefile

```diff
@@ -14,8 +14,17 @@ help:
 -include $(HOME)/.config/stan/make.local  # user-defined variables
 -include make/local                       # user-defined variables
 
-MATH ?= lib/stan_math/
-RAPIDJSON ?= lib/rapidjson_1.1.0/
+ifdef MATH_INC
```

Comment:
@andrjohns, I believe the existing code does what you want and is cleaner makefile code. 

In your make/local, if you change your example to use the existing variables, it should work. 

Does it not? If not, we can trace why it's not picking up on changes made at the make/local level. 

* could you also explain the reason for duplicating the variables?
* could you give some context to the names? (Why _INC instead of something more descriptive if we're going that way? Include is ambiguous as a term here... source include or library include?)

Reply:
This is for consistency with the existing approach with providing external headers for the TBB in the Math library (described in the README [here](https://mc-stan.org/math#title3)), where the `_INC` suffix denotes the path to header includes and the `_LIB` suffix denotes the path to shared libraries (i.e., dll/so/dylib).

I felt that having the `_INC` suffix for the math library and rapidjson made for a unified approach (as it is also being used in the linked Math library PR for the other dependencies)

Reply:
Hmm but after thinking about it more, if the current makefiles are already compatible then it's unnecessary to be doing this. I'll close this PR and instead just check that the current makefiles work without issue - and open a PR with any fixes needed if not